docs/docs/workflows/terraform-module.md

```diff
@@ -0,0 +1,130 @@
+    name    = "gcp-ex"
+    project = "constellation-331613"
+    service_account_id = "ex"
+    image   = "$VERSION" // replace with the desired version, e.g., `v2.13.0`.
+    zone    = "europe-west2-a"
+    node_groups = {
+      control_plane_default = {
+        role          = "control-plane"
+        zone          = "europe-west2-a"
+        instance_type = "n2d-standard-4"
+        disk_size     = 30
+        disk_type     = "pd-ssd"
+        initial_count = 3
+      },
+      worker_default = {
+        role          = "worker"
+        zone          = "europe-west2-a"
+        instance_type = "n2d-standard-4"
+        disk_size     = 30
+        disk_type     = "pd-ssd"
+        initial_count = 2
+      }
+    }
+  }
+  ```
+
+  </tabItem>
+
+  </tabs>
+
+3. Initialize and apply Terraform.
+  ```bash
+  terraform init
+  terraform apply
+  ```
+
+## Custom setup
+If you need to separate IAM and cluster management or need custom infrastructure, you can also call the submodules individually.
+Look at the respective convenience module (`{csp}-constellation`) for the structure.
+The submodules are:
+- `constellation-cluster`: fetches the CLI, creates the Constellation configuration, and applies it to the cluster
+- `fetch-image`: translates the Constellation image version to the image ID of the cloud service provider
+- `infrastructure/{csp}`: contains the cluster infrastructure resources
+- `infrastructure/iam/{csp}`: contains the resources to set up IAM
+
+## Cluster upgrades
+The usage of a [remote URL as module source](https://developer.hashicorp.com/terraform/language/modules/sources#fetching-archives-over-http) is recommended, because it simplifies the upgrade process. For [local paths as module source](https://developer.hashicorp.com/terraform/language/modules/sources#local-paths), the user needs to manually overwrite the Terraform files in the Terraform workspace. The steps for the remote URL setup are as follows:
+
+1. Update the `source` field by inserting the `$NEW_VERSION`.
+    ```
+        source = "https://github.com/edgelesssys/constellation/releases/download/$NEW_VERSION/terraform-module.zip//modules/$CSP-constellation" # TODO check the link during the release.
+   ```
+2. (Optional) Update the image version.
+
+    `image` is a mandatory field and needs to be updated to `$NEW_VERSION` to perform a node image upgrade.
+
+3. (Optional) Update the Kubernetes version.
+
+    If `kubernetes_version` isn't set, the latest default version of the newer Constellation release will be used, which may imply a Kubernetes upgrade. If you want to avoid implicit upgrades, explicitly set this field. For the supported versions, refer to [Kubernetes support policy](../architecture/versions.md#kubernetes-support-policy).
```

Comment:
```suggestion
    If `kubernetes_version` isn't set, the latest default version of the newer Constellation release will be used, which may imply a Kubernetes upgrade. If you want to avoid implicit upgrades, explicitly set this field. For the supported versions, refer to [ Constellation's Kubernetes support policy](../architecture/versions.md#kubernetes-support-policy).
```